ui/src/utils/env.ts

```diff
@@ -0,0 +1,47 @@
+export {}
+
+const GIT_SHA =
+  process.env.INFLUXDB_SHA ||
+  require('child_process')
+    .execSync('git rev-parse HEAD')
+    .toString()
+
+const STATIC_DIRECTORY = (dir => {
+  if (!dir.length) {
+    return dir
+  }
+
+  return (
+    dir.slice(
+      dir[0] === '/' ? 1 : 0,
+      dir[dir.length - 1] === '/' ? -1 : undefined
+    ) + '/'
+  )
+})(process.env.STATIC_DIRECTORY || '')
+
+const BASE_PATH = (prefix => {
+  // TODO: adding a basePath feature in the @influxdata/oats
+  // project is currently required before turning this on
+  // just remove this return when that is done (alex)
+  return '/'
+
+  /* eslint-disable no-unreachable */
+  if (prefix === '/') {
+    return prefix
+  }
+
+  return prefix[0] === '/'
+    ? prefix[prefix.length - 1] === '/'
+      ? prefix
+      : prefix + '/'
+    : prefix[prefix.length - 1] === '/'
+    ? '/' + prefix
+    : '/' + prefix + '/'
```

Comment:
it's probably cause i'm dumber than hell, but this takes a lot of brainpower for me to follow, between the nested ternaries and the length calculations to figuring out where in the string it is. (it also looks like there are identical cases for for the first nesting of `?` and `:` and i can't quite figure it out - again, dumb). do you think you could convert this to a regular `if / else` block or maybe introduce some variables to store intermediate results to make it a little bit easier to read?

Reply:
yeah, was conflicted about this as well. i saw a nested ternary or two in the codebase, but this one was loooooong and made it that much harder to read. update forthcoming